onnx/defs/function.h

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Facebook Inc. and Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+#include <mutex>
+#include <string>
+#include <vector>
+
+#include "onnx/common/status.h"
+#include "onnx/onnx-operators_pb.h"
+
+namespace ONNX_NAMESPACE {
+using namespace Common;
+
+typedef Common::Status (*BuildFunction)(std::unique_ptr<FunctionProto>*);
```

Comment:
Could you use `FunctionProto*` here?

Reply:
I'd keep the unique_ptr here to 1) avoid a copy, 2) clarify its unique ownership (folks should not copy it over around).